packages/react-magma-dom/src/components/AIButton/AIButton.stories.tsx

```diff
@@ -0,0 +1,232 @@
+  AIButtonProps,
+  AIButton,
+  AIButtonVariant,
+  AIButtonSize,
+  AIButtonShape,
+  AIButtonTextTransform,
+  AIButtonType,
+} from '.';
+
+const Template: Story<AIButtonProps> = args => (
+  <Card isInverse={args.isInverse}>
+    <CardBody>
+      <ButtonGroup>
+        <AIButton {...args}>Variant A</AIButton>
+        <AIButton variant={AIButtonVariant.variantB} {...args}>
+          Variant B
+        </AIButton>
+      </ButtonGroup>
+      <Spacer size={16} />
+      <ButtonGroup>
+        <AIButton {...args} />
+        <AIButton variant={AIButtonVariant.variantB} {...args} />
+      </ButtonGroup>
+    </CardBody>
+  </Card>
+);
+
+export default {
+  title: 'AIButton',
+  component: AIButton,
+  argTypes: {
+    size: {
+      control: {
+        type: 'select',
+        options: AIButtonSize,
+      },
+    },
+    variant: {
+      control: {
+        type: 'select',
+        options: AIButtonVariant,
+      },
+    },
+    shape: {
+      control: {
+        type: 'select',
+        options: AIButtonShape,
+      },
+    },
+    textTransform: {
+      control: {
+        type: 'select',
+        options: AIButtonTextTransform,
+      },
+    },
+    type: {
+      control: {
+        type: 'select',
+        options: AIButtonType,
+      },
+    },
+    isLoading: {
+      control: {
+        type: 'boolean',
+      },
+    },
+    isAnimated: {
+      control: {
+        type: 'boolean',
+      },
+    },
+    leftColor: {
+      control: {
+        type: 'color',
+      },
+    },
+    rightColor: {
+      control: {
+        type: 'color',
+      },
+    },
+    hoverColor: {
+      control: {
+        type: 'color',
+      },
+    },
+    pressedColor: {
+      control: {
+        type: 'color',
+      },
+    },
+    onClick: { action: 'clicked' },
+  },
+} as Meta;
+
+export const Default = Template.bind({});
+Default.args = {
+  isInverse: false,
+  isFullWidth: false,
+  disabled: false,
+};
+
+export const LoadingButton = args => {
+  const [isLoading, setIsLoading] = React.useState(args.isLoading);
+  const [isLoadingIcon, setIsLoadingIcon] = React.useState(args.isLoading);
+
+  React.useEffect(() => {
+    if (isLoading) {
+      setTimeout(() => {
+        setIsLoading(false);
+      }, 4000);
+    }
+    if (isLoadingIcon) {
+      setTimeout(() => {
+        setIsLoadingIcon(false);
+      }, 4000);
+    }
+  }, [isLoading, isLoadingIcon]);
+
+  return (
+    <>
+      <p>Click the AIButton below to show the loading state</p>
+      <VisuallyHidden>
+        <span role="status">{isLoading ? 'Loading...' : 'Ready'}</span>
+      </VisuallyHidden>
+      <ButtonGroup>
+        <AIButton
+          {...args}
+          isLoading={isLoading}
+          onClick={() => setIsLoading(true)}
+        >
+          Save
+        </AIButton>
+        <AIButton
+          {...args}
+          isLoading={isLoadingIcon}
+          onClick={() => setIsLoadingIcon(true)}
+        />
+      </ButtonGroup>
+    </>
+  );
+};
+
```

Comment:
```suggestion
LoadingButton.args = {
  ...Default.args,
};
```

Reply:
I also just noticed that clicking the loading button with text changes the width of the button (we don't see this behavior in the regular button)